File: src/android/KotlinSample/app/src/main/java/com/barry/sample/kotlin/MainActivity.java
package com.barry.sample.kotlin;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    Button btnHello = (Button) findViewById(R.id.btn_hello);
    if (btnHello != null) {
      btnHello.setOnClickListener(new View.OnClickListener() {
        @Override public void onClick(View v) {
          TextView txtHello = (TextView) findViewById(R.id.txt_hello);
          EditText txtName = (EditText) findViewById(R.id.txt_name);
          if ((txtHello != null) && (txtName != null)) {
            String hello_sentence =
                String.format(getString(R.string.hello_sentence), txtName.getText());
            txtHello.setText(hello_sentence);
          }
        }
      });
    }
  }
}

File: src/android/KotlinSample/app/src/main/kotlin/com/barry/sample/kotlin/MainActivity.kt
package com.barry.sample.kotlin

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {

  // Lambda Expression
  val sayHello: (String) -> String = { text ->
    String.format(getString(R.string.hello_sentence), text.capitalize())
  }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    val btnHello = findViewById(R.id.btn_hello) as Button?
    btnHello?.setOnClickListener {
      val txtHello = findViewById(R.id.txt_hello) as TextView?
      val txtName = findViewById(R.id.txt_name) as EditText?
      if (txtHello != null && txtName != null) {
        txtHello.text = applyFormat(txtName.text.toString(), sayHello)
      }
    }
  }

  fun applyFormat(text: String, f: (String) -> String): String {
    return f(text)
  }
}


